### ISSUE TITLE
Unsure how to use SqliteQueuedDatabase across multiple processes

### ISSUE THREAD

**Issue by NickM-27** [coleifer/peewee#2740]
Hi, thank you for this awesome project. I hope I can paint a clear picture and get some advice on the best way to proceed.

I contribute to a python application which used a `SqliteQueuedDatabase` and the `Database.bind([Models])` function in a multi-threaded but single process manner with great success. Due to unrelated issues we needed to break out part of the application's logic into a separate process (mp.Process) which has caused issues. I'll outline the following attempts and issue that was encountered:

1. Create separate SqliteQueuedDatabase

I tried starting the second process, create a SqliteQueuedDatabase & bind() in the second process, then create a SqliteQueuedDatabase & bind() in the main process. This works for the most part but I've started encountering `db is locked` I'm assuming because the databases have separate queues and are unaware of each other which can lead to write conflicts.

2. Only use a single SqliteQueuedDatabase

I tried just using a single SqliteQueuedDatabase & bind() in the main process however it seems that the DB operations from the models in the second process never get completed. 

To summarize, the ideal solution would be to have things continue to work as they did in a single process but with multiple processes. I appreciate any suggestions and advice, thanks!

**Comment by coleifer** [comment]
You will need to rearchitect things fairly significantly if you want to use this with multiple processes. The design is informed by the fact that we can share memory across threads and have full visibility into who is doing what. Not sure what platform you're on, but whether your multiprocessing is using `fork` or `spawn` for the child processes will also have an impact - I suggest reading up on the differences (I'm guessing fork just based on the behavior you describe).

In order to use multiple processes, you will probably want to maintain a single, dedicated process for performing the write operations and then as many processes as you wish for the cpu-intensive work. You might try avoiding using the `SqliteQueueDatabase` in the first place when working with multiple processes. Also ensure that when you start new processes the database is in a closed state and you establish your connection at the beginning.

There are far too many caveats and details than I can hope to cover.

**Comment by NickM-27** [comment]
Thank you for the answer, a specific process dedicated to writing was one idea so it seems that's the way we'll want to go.
